utils/search/backends/test_search_engine_backend.py

```diff
@@ -246,6 +249,44 @@ def sound_check_extra_queries(self):
         self.run_sounds_query_and_save_results(dict(textual_query='piano loop'))
         self.run_sounds_query_and_save_results(dict(textual_query='explosion'))
 
+    def sound_check_get_user_tags(self, sound):
+        """
+        search_engine.get_user_tags() returns the top 10 tags that a user has applied,
+        so this could by chance not coincide with any of the tags on this sound.
+        Instead, get all tags by the user and check that the ones from solr are a subset of them.
+        """
+        user_tagged_items = TaggedItem.objects.filter(user=sound.user).select_related('tag').all()
+        all_user_tags = [ti.tag.name for ti in user_tagged_items]
+        tags_and_counts = self.search_engine.get_user_tags(sound.user.username)
+        solr_tags = [t[0] for t in tags_and_counts]
+
+        remaining_tags = set(solr_tags) - set(all_user_tags)
+        assert_and_continue(len(remaining_tags) == 0, "get_user_tags returned tags which the user hasn't tagged")
```

Comment:
we don't write anything to the test execution file here, I'm open to ideas for suggestions about what we should write. Perhaps the value of these two sets? The sound id/name?

Reply:
The sets of tags could be nice